client/src/components/Toolshed/RepositoryDetails/Index.vue

```diff
@@ -1,3 +1,228 @@
+
+import { Services } from "../services";
+
+import InstallationActions from "./InstallationActions.vue";
+import InstallationSettings from "./InstallationSettings.vue";
+import RepositoryTools from "./RepositoryTools.vue";
+
+const props = defineProps({
+    repo: {
+        type: Object,
+        required: true,
+    },
+    toolshedUrl: {
+        type: String,
+        required: true,
+    },
+});
+
+const services = new Services();
+
+const { config } = useConfig(true);
+const { panel, fetchPanel } = useToolStore();
+
+const repositoryWatcher = useResourceWatcher(loadInstalledRepositories, {
+    shortPollingInterval: 2000,
+    enableBackgroundPolling: false,
+});
+
+const repoChecked = "fa fa-check text-success";
+const repoUnchecked = "fa fa-times text-danger";
+const repoFields = [
+    { key: "numeric_revision", label: "Revision" },
+    { key: "tools", label: "Tools and Versions" },
+    { key: "profile", label: "Requires" },
+    { key: "missing_test_components", label: "Tests" },
+    { key: "status" },
+    { key: "actions", label: "", class: "toolshed-repo-actions" },
+];
+
+const statusError = "Error";
+const statusInstalled = "Installed";
+const statusUninstalled = "Uninstalled";
+
+const revisionStateMap = new Map();
+const revisionWaitStateMap = new Map();
+
+const selectedChangeset = ref();
+const selectedRequiresPanel = ref(false);
+const repoTable = ref([]);
+const showSettings = ref(false);
+const error = ref();
+const loading = ref(true);
+
+onMounted(() => {
+    load();
+    if (!panel["default"]) {
+        fetchPanel("default");
+    }
+});
+
+onUnmounted(() => {
+    stopWatchingRepository();
+});
+
+async function load() {
+    try {
+        const response = await services.getRepository(props.toolshedUrl, props.repo.id);
+        repoTable.value = response ?? [];
+        loadInstalledRepositories();
+    } catch (e) {
+        error.value = errorMessageAsString(e);
+        loading.value = false;
+    }
+}
+
+async function loadInstalledRepositories() {
+    try {
+        const revisions = await services.getInstalledRepositoriesByName(props.repo.name, props.repo.owner);
+
+        refreshRepositoryStatus(revisions);
+
+        if (!isAnyChangeExpected()) {
+            stopWatchingRepository();
+        }
+    } catch (e) {
+        error.value = errorMessageAsString(e);
+    } finally {
+        loading.value = false;
+    }
+}
+
+function refreshRepositoryStatus(updatedRevisions) {
+    repoTable.value.forEach((x) => {
+        const repoRevision = updatedRevisions[x.changeset_revision];
+        if (!repoRevision) {
+            return;
+        }
+        revisionStateMap.set(x.changeset_revision, repoRevision);
+        if (repoRevision.status !== x.status) {
+            x.status = repoRevision.status;
+            x.installed = repoRevision.installed;
+        }
+    });
+
+    repoTable.value = [...repoTable.value];
+}
+
+function isAnyChangeExpected() {
```

Comment:
I think we can replace this with a constant of true ? 😆 
We're always expecting changes, from another tab or the API.

Reply:
If we're trying to reduce polling frequency, could we instead cycle through the list of repositories, or filter `/api/tool_shed_repositories` by update_time, or filter `api/tool_shed_repositories` by the expanded repository ids ? But is this really such a problem that it requires addressing ? Even if you expand 100 repositories that doesn't really seem like it should be a problem.

Reply:
Yeah, for sure, the changes look solid to me, but I think I'd prefer being a little dumber here and poll for anything that is expanded.